azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/AbstractAppServiceCreator.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.toolkits.appservice.service;
+
+import com.microsoft.azure.toolkits.appservice.entity.AppServicePlanEntity;
+import com.microsoft.azure.toolkits.appservice.model.DockerConfiguration;
+import com.microsoft.azure.toolkits.appservice.model.PricingTier;
+import com.microsoft.azure.toolkits.appservice.model.Runtime;
+import com.microsoft.azure.tools.common.model.Region;
+import lombok.Getter;
+
+import java.util.Map;
+import java.util.Optional;
+
+@Getter
+public abstract class AbstractAppServiceCreator<T> implements IAppServiceCreator<T> {
```

Comment:
Double confirm to do like this for "@Getter". It maybe not skip these public/static properties. I prefer to code it above these private properties we need one by one.